extensions/flyway/deployment/src/main/java/io/quarkus/flyway/FlywayProcessor.java

```diff
@@ -121,15 +120,19 @@ private void registerNativeImageResources(BuildProducer<NativeImageResourceBuild
             throws IOException, URISyntaxException {
         List<String> resources = new ArrayList<>();
         try {
-            List<String> locations = new ArrayList<>(flywayBuildConfig.locations.orElse(Collections.emptyList()));
-            if (locations.isEmpty()) {
-                locations.add("db/migration");
+            String[] defaultMigrationLocation = { "db/migration" };
+            String[] locations = flywayBuildConfig.locations.orElse(defaultMigrationLocation);
+            if (locations.length == 0) {
+                locations = defaultMigrationLocation;
             }
             // Locations can be a comma separated list
             for (String location : locations) {
+                if (location == null) {
```

Comment:
Is this check required?

Reply:
I am not 100% sure. it was there before see old line 132 below

Reply:
if it is not possible get `null` from a Config, it could be removed.

Reply:
It is indeed not possible to get `null` from a `Config`.